src/index.ts

```diff
@@ -64,13 +64,34 @@ export const submitTask = async (dataId: string, dataUserPk: string, wallet: any
   return taskId;
 }
 
-export const getResult = async (taskId: string, dataUserSk: string,
-  arweave: Arweave = Arweave.init({})) => {
-  const taskStr = await getCompletedTasksById(taskId);
+/**
+ * Get the result of the task
+ * @param taskId taskId
+ * @param timeout Timeout in milliseconds
+ */
+const getResultPromise = (taskId: string, timeout: number): Promise<string> => {
+  return new Promise((resolve, reject) => {
+    const start = performance.now();
+    const tick = async () => {
+      const timeGap = performance.now() - start;
+      const taskStr = await getCompletedTasksById(taskId);
+      const task = JSON.parse(taskStr);
+      if (task.id) {
+        resolve(taskStr);
+      } else if (timeGap > timeout) {
+        reject('timeout');
+      } else {
+        setTimeout(tick, 500);
+      }
+    };
+    tick();
+  });
+};
+
+export const getResult = async (taskId: string, dataUserSk: string, timeout: number,
```

Comment:
the timeout params can assign a default value

Reply:
The timeout parameter has been changed to a non-required parameter, with a default of 10 seconds.